sentry/src/main/java/io/sentry/RequestDetailsResolver.java

```diff
@@ -4,24 +4,39 @@
 import java.net.URI;
 import java.util.HashMap;
 import java.util.Map;
+import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 /** Resolves {@link RequestDetails}. */
-final class RequestDetailsResolver {
+@ApiStatus.Experimental
+public final class RequestDetailsResolver {
   /** HTTP Header for the user agent. */
   private static final String USER_AGENT = "User-Agent";
   /** HTTP Header for the authentication to Sentry. */
   private static final String SENTRY_AUTH = "X-Sentry-Auth";
 
-  private final @NotNull SentryOptions options;
+  private final @NotNull Dsn dsn;
+  private final @Nullable String sentryClientName;
 
+  public RequestDetailsResolver(
+      final @NotNull String dsn, final @Nullable String sentryClientName) {
```

Comment:
Could technically do without this constructor, make the `SentryOptions` one public, and pass fake `SentryOptions` with only dsn and client name from the caller side. Disadvantage is it would break if any other options are used in the future. (but disadvantage of this constructor is that adding a parameter would be breaking API change)